From 8c8f6c2f931cd8722dc4c03c70e0a4d32d921acd Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Wed, 29 Oct 2025 13:43:15 +0800
Subject: [PATCH 231/239] Dial::test_integerStepping: ensure we actually test
 changes in value

By not comparing the Dial's value against 1 eight times.

Amends 42687caf7bc7496b838995f3fa68194ca8323da3.

Pick-to: 6.8
Change-Id: I67bc64c517d06a176ba4b2189f5755f954915ae1
Reviewed-by: Oliver Eftevaag <oliver.eftevaag@qt.io>
(cherry picked from commit 089508cd4ea9a76892a183ba4f070325c1add4ac)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/quickcontrols/controls/data/tst_dial.qml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/auto/quickcontrols/controls/data/tst_dial.qml b/tests/auto/quickcontrols/controls/data/tst_dial.qml
index 1f2b9fdd5c..b7c7c2ad97 100644
--- a/tests/auto/quickcontrols/controls/data/tst_dial.qml
+++ b/tests/auto/quickcontrols/controls/data/tst_dial.qml
@@ -659,10 +659,12 @@ TestCase {
         dial.from = 1
         dial.to = 8
         dial.stepSize = 1
+        dial.forceActiveFocus()
+        verify(dial.activeFocus)
 
         for (let i = 1; i < 8; ++i) {
             // compare as strings to avoid a fuzzy compare; we want an exact match
-            compare(""+dial.value, ""+1)
+            compare(""+dial.value, ""+i)
             keyClick(Qt.Key_Right)
         }
     }
-- 
2.51.2

